cli: TestPartialZip failed

cli.TestPartialZip [failed](https://mesolite.cluster.engflow.com/invocations/default/4aedefe0-7695-48f8-82ad-2f334a76e304?testReportRun=2&testReportShard=14&testReportAttempt=1#targets-Ly9wa2cvY2xpOmNsaV90ZXN0) on release-25.3 @ [e85eeea015cec27a716de792c527dd83cabcf0ab](https://github.com/cockroachdb/cockroach/commits/e85eeea015cec27a716de792c527dd83cabcf0ab):


```
          [node 3] requesting data for debug/nodes/3/gossip... received response... writing JSON output: debug/nodes/3/gossip.json... done
          [node 3] requesting stacks... received response... writing binary output: debug/nodes/3/stacks.txt... done
          [node 3] requesting stacks with labels... received response... writing binary output: debug/nodes/3/stacks_with_labels.txt... done
          [node 3] requesting heap profile... received response... writing binary output: debug/nodes/3/heap.pprof... done
          [node 3] requesting engine stats... received response... writing binary output: debug/nodes/3/lsm.txt... done
          [node 3] requesting heap profile list... received response...
          [node 3] requesting heap profile list: last request failed: rpc error: ...
          [node 3] requesting heap profile list: creating error output: debug/nodes/3/heapprof.err.txt... done
          [node 3] requesting goroutine dump list... received response...
          [node 3] requesting goroutine dump list: last request failed: rpc error: ...
          [node 3] requesting goroutine dump list: creating error output: debug/nodes/3/goroutines.err.txt... done
          [node 3] requesting cpu profile list... received response...
          [node 3] requesting cpu profile list: last request failed: rpc error: ...
          [node 3] requesting cpu profile list: creating error output: debug/nodes/3/cpuprof.err.txt... done
          [node 3] requesting execution trace list... received response...
          [node 3] requesting execution trace list: last request failed: rpc error: ...
          [node 3] requesting execution trace list: creating error output: debug/nodes/3/executiontraces.err.txt... done
          [node 3] requesting log files list... received response... done
          [node ?] ? log files found
          [node 3] requesting ranges... received response... done
          [node 3] writing ranges... writing JSON output: debug/nodes/3/ranges.json... done
          [cluster] pprof summary script... writing binary output: debug/pprof-summary.sh... done
          [cluster] hot range summary script... writing binary output: debug/hot-ranges.sh... done
          [cluster] tenant hot range summary script... writing binary output: debug/hot-ranges-tenant.sh... done
          [cluster] capture debug zip flags... writing binary output: debug/debug_zip_command_flags.txt... done
[debug --host=127.0.0.1:38613 --insecure=true zip --concurrency=1 --exclude-nodes=2 --cpu-profile-duration=0 --validate-zip-file=false --cluster-name=TestCluster-5SIcIQww3h /dev/null]
    zip_test.go:770: 
          /mnt/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/cli/cli_test_/cli_test.runfiles/com_github_cockroachdb_cockroach/pkg/cli/testdata/zip/partial1_excluded:1:
          zip
        output didn't match expected:
          @@ -5,10 +5,15 @@
           [cluster] using SQL address: ...
           [cluster] requesting data for debug/events... received response... writing JSON output: debug/events.json... done
           [cluster] requesting data for debug/rangelog... received response... writing JSON output: debug/rangelog.json... done
           [cluster] requesting data for debug/settings... received response... writing JSON output: debug/settings.json... done
           [cluster] requesting data for debug/reports/problemranges... received response... writing JSON output: debug/reports/problemranges.json... done
          +<dumping SQL tables>
          +[cluster] retrieving SQL data for system.role_options... writing output: debug/system.role_options.txt...NOTICE: No license is installed. Throttling will begin after 2025-12-03 09:08:41 +0000 UTC unless a license is installed before then.
          +NOTICE: No license is installed. Throttling will begin after 2025-12-03 09:08:41 +0000 UTC unless a license is installed before then.
          +NOTICE: No license is installed. Throttling will begin after 2025-12-03 09:08:41 +0000 UTC unless a license is installed before then.
          + done
           <dumping SQL tables>
           [cluster] requesting nodes... received response... writing JSON output: debug/nodes.json... done
           [cluster] requesting liveness... received response... writing JSON output: debug/liveness.json... done
           [cluster] requesting tenant ranges... received response...
           [cluster] requesting tenant ranges: last request failed: rpc error: ...
    panic.go:635: -- test log scope end --
    panic.go:635: -- test log scope end --
test logs left over in: outputs.zip/logTestPartialZip1573419465
--- FAIL: TestPartialZip (29.41s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=14</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #154424 cli: TestPartialZip failed [C-test-failure O-robot T-supportability branch-master release-blocker]
</p>
</details>
/cc @cockroachdb/obs-prs @cockroachdb/server @cockroachdb/obs-india-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestPartialZip.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-57260

--- a/pkg/cli/zip_test.go
+++ b/pkg/cli/zip_test.go
@@ -718,6 +718,7 @@ func TestPartialZip(t *testing.T) {
 	// however low timeouts make race runs flaky with false positives.
 	skip.UnderShort(t)
 	skip.UnderRace(t)
+	skip.UnderDeadlock(t, "flaky under deadlock")
 
 	sc := log.ScopeWithoutShowLogs(t)
 	defer sc.Close(t)
@@ -830,6 +831,7 @@ func TestZipDisallowFullScans(t *testing.T) {
 
 	skip.UnderShort(t)
 	skip.UnderRace(t)
+	skip.UnderDeadlock(t, "flaky under deadlock")
 
 	dir, cleanupFn := testutils.TempDir(t)
 	defer cleanupFn()
@@ -1500,6 +1502,8 @@ func trimNonDeterministicZipOutputFiles(out string) string {
 func TestZipIncludeAndExcludeFilesDataDriven(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
+
+	skip.UnderDeadlock(t, "flaky under deadlock")
 
 	datadriven.Walk(t, "testdata/zip/file-filters", func(t *testing.T, path string) {
 		datadriven.RunTest(t, path, func(t *testing.T, d *datadriven.TestData) string {

